### DISCUSSION TITLE
How to calc train_accuracy on end of every epoch [beginner]

### DISCUSSION THREAD

**Discussion by Muthukamalan** [Lightning-AI/pytorch-lightning#19749] [code help: CV]
Hi,

objective:  I want to calculate accuracy of training set only on epoch count less than 3 and epoch%50==0
kind of,
```python
class MNISTNet(pl.LightningModule):
  def training_epoch_end(self, batch,batch_idx):
    if self.trainer.current_epoch <3 or self.trainer.current_epoch%50==0:
      x,y = batch
      out = self(x)   
      acc = acc(x,out)
```

after I googled,
>>>  def on_train_batch_end(self,outputs): pass
>>> def on_train_epoch_end(self): pass 

with this may I achieve my objective and I don't understand outputs here?
